### ISSUE TITLE
Type promotion errors in grad of some nn functions with numpy==1.23.0

### ISSUE THREAD

**Issue by seanmor5** [google/jax#11286]
I'm experiencing some type promotion errors after upgrading to `numpy==1.23.0`. I'm guessing it's probably not fully supported yet---my current JAX version is `0.3.13`.

The following is a basic repro to trigger the error:

```python
import jax
import jax.numpy as jnp

jax.grad(lambda x: jnp.sum(jax.nn.softplus(x)))(jnp.array([1.0, 2.0, 3.0]))
```

Filtered stack trace:

```
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "jax_bug.py", line 4, in <module>
    jax.grad(lambda x: jnp.sum(jax.nn.softplus(x)))(jnp.array([1.0, 2.0, 3.0]))
  File "jax_bug.py", line 4, in <lambda>
    jax.grad(lambda x: jnp.sum(jax.nn.softplus(x)))(jnp.array([1.0, 2.0, 3.0]))
  File "/home/sean/.local/lib/python3.8/site-packages/jax/_src/nn/functions.py", line 66, in softplus
    return jnp.logaddexp(x, 0)
  File "/home/sean/.local/lib/python3.8/site-packages/jax/_src/numpy/ufuncs.py", line 361, in _logaddexp_jvp
    x1, x2, t1, t2 = _promote_args_inexact("logaddexp_jvp", x1, x2, t1, t2)
  File "/home/sean/.local/lib/python3.8/site-packages/jax/_src/numpy/util.py", line 327, in _promote_args_inexact
    return _promote_shapes(fun_name, *_promote_dtypes_inexact(*args))
  File "/home/sean/.local/lib/python3.8/site-packages/jax/_src/numpy/util.py", line 262, in _promote_dtypes_inexact
    to_dtype, weak_type = dtypes._lattice_result_type(*args)
KeyError: dtype([('float0', 'V')])
```

Note from the stacktrace I thought the culprit would be from `logaddexp`, so I tried:

```python                                                                                
import jax
import jax.numpy as jnp

jax.grad(lambda x: jnp.sum(jnp.logaddexp(x, 0)))(jnp.array([1.0, 2.0, 3.0]))
```

which is the exact implementation of `jax.nn.softplus` and it runs without issue

Also note that downgrading to `numpy==1.22.3` gets rid of the error completely

**Comment by jakevdp** [comment]
Thanks - this is a duplicate of https://github.com/google/jax/issues/11221, and has been fixed in https://github.com/google/jax/pull/11230. As you mentioned, until the next jax release is available, the best fix is to downgrade numpy to version 1.22.4 or older.

**Comment by seanmor5** [comment]
Ah sorry for the dupe! I was looking in the wrong place. Thanks!
